reporter/sbom/spdx.go

```diff
@@ -0,0 +1,555 @@
+package sbom
+
+import (
+	"cmp"
+	"encoding/json"
+	"fmt"
+	"math/rand/v2"
+	"slices"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/spdx/tools-golang/spdx"
+	"github.com/spdx/tools-golang/spdx/v2/common"
+
+	"github.com/future-architect/vuls/config"
+	"github.com/future-architect/vuls/constant"
+	"github.com/future-architect/vuls/models"
+)
+
+const (
+	creatorOrganization = "future-architect"
+	creatorTool         = "vuls"
+	documentNamespace   = "https://www.future.co.jp/spdxdoc"
+
+	documentSPDXIdentifier = "DOCUMENT"
+	elementOperatingSystem = "Operating-System"
+	elementPackage         = "Package"
+
+	packagePurposeOS          = "OPERATING-SYSTEM"
+	packagePurposeApplication = "APPLICATION"
+	packagePurposeLibrary     = "LIBRARY"
+
+	packageAnnotatorTool = "Tool"
+	annotationOther      = "Other"
+
+	valueNone = "NONE"
+
+	relationshipDescribe = common.TypeRelationshipDescribe
+	relationshipContains = common.TypeRelationshipContains
+	relationshipDepensOn = common.TypeRelationshipDependsOn
+
+	categoryPackageManager = common.CategoryPackageManager
+	packageManagerPURL     = common.TypePackageManagerPURL
+
+	categorySecurity  = common.CategorySecurity
+	securityCPE23Type = common.TypeSecurityCPE23Type
+	securityAdvisory  = common.TypeSecurityAdvisory
+)
+
+// ToSPDX converts ScanResult to SPDX Document
+func ToSPDX(r models.ScanResult, toolName string) spdx.Document {
+	root := osToSpdxPackage(r)
+
+	packageToURLMap := createPackageToURLMap(r)
+	creationInfo := spdxCreationInfo(r, toolName)
+	packages, relationships := spdxPackages(r, root, packageToURLMap)
```

Comment:
packageToURLMap is only used this function call, and the argument of createPackageToURLMap() "r" is also passed to spdxPackages().
It may indicate that it's better to call createPackageToURLMap() inside of spdxPackages()?